addon/number.js

```diff
@@ -30,16 +30,20 @@ import validationError from 'ember-validators/utils/validation-error';
 export default function validateNumber(value, options) {
   let numValue = Number(value);
   let optionKeys = Object.keys(options);
-  let { allowBlank, allowNone = true, allowString, integer } = getProperties(options, ['allowBlank', 'allowNone', 'allowString', 'integer']);
+  let { allowBlank, allowNone, allowString, integer } = getProperties(options, ['allowBlank', 'allowNone', 'allowString', 'integer']);
 
-  if (!allowNone && isNone(value)) {
-    return validationError('notANumber', value, options);
+  if (allowNone && isNone(value)) {
+    return true;
   }
 
   if (allowBlank && isEmpty(value)) {
     return true;
   }
 
+  if (isNone(value)) {
+    return validationError('notANumber', value, options);
+  }
```

Comment:
Note that I had to move this below the `allowBlank` check, otherwise it would break when `allowBlank = true` and the value is `null`/`undefined`.